### DISCUSSION TITLE
Fermionic Operators in SU2

### DISCUSSION THREAD

**Discussion by cfengno1** [block-hczhai/block2-preview#146] [Q&A] [answered]
Hi Huanchen,

In then get_npdm() defined in core.py:
``
if SymmetryTypes.SU2 in bw.symm_type:
            assert fermionic_ops is None
``
Why can't one use fermionic_ops in the SU2 symmetry? Can one use define fermionic_ops in the construction of MPO of the Hamiltonian?

Best,
Feng

**Comment by hczhai** [comment, marked as answer]
> Can one use define fermionic_ops in the construction of MPO of the Hamiltonian?

When you set ``adjust_order=False`` when building the MPO, the value of ``fermionic_ops`` will not be used so it does not matter.

> Why can't one use fermionic_ops in the SU2 symmetry?

This is related to the comment in https://github.com/block-hczhai/block2-preview/discussions/139#discussioncomment-11125818. In the SU2 mode, we do not allow custom operators, so that the ``block2`` code knows which operators are fermionic and which are not. So ``fermionic_ops=None`` means it is not user's responsibility for defining the fermionicity in the SU2 mode.

In ``get_npdm()``, the operator index for computing expectation values will inevitably have non-ascending ordering, so you can understand it as there is an enforced implicit ``adjust_order=True`` in ``get_npdm()``.
